File: spring-email-java/src/main/java/com/github/migangqui/spring/email/service/EmailService.java
package com.github.migangqui.spring.email.service;

import com.github.migangqui.spring.email.bean.SendEmailResult;
import com.github.migangqui.spring.email.bean.Email;

import java.util.concurrent.Future;

public interface EmailService {
    SendEmailResult send(Email emailDTO);
    
    Future<SendEmailResult> sendAsync(Email emailDTO);
}

File: spring-email-kotlin/src/main/kotlin/com/github/migangqui/spring/email/service/EmailService.kt
package com.github.migangqui.spring.email.service

import com.github.migangqui.spring.email.bean.Email
import com.github.migangqui.spring.email.bean.SendEmailResult
import java.util.concurrent.Future

interface EmailService {
    fun send(email: Email): SendEmailResult

    fun sendAsync(email: Email): Future<SendEmailResult>
}